collect_app/src/main/java/org/odk/collect/android/storage/migration/StorageMigrator.java

```diff
@@ -104,11 +107,11 @@ public StorageMigrationResult migrate() {
     }
 
     private boolean isFormUploaderRunning() {
-        return backgroundWorkManager.isFormUploaderRunning();
+        return formSubmitManager.isSubmitRunning();
```

Comment:
I think really we just need a shared "lock" in the app that stops things that make changes running at the same time as here we have to care about multiple things, not everything is covered and there is no real thread safety. I've noted this down and I want to work on it later once we've added error handling for match exactly.